vllm/v1/engine/core.py

```diff
@@ -95,14 +96,14 @@ def _initialize_kv_caches(self,
 
     def add_request(self, request: EngineCoreRequest):
         """Add request to the scheduler."""
-
         req = Request.from_engine_core_request(request)
-        # FIXME(woosuk): The input mapping (e.g., PIL images to tensors) may
-        # take 10-50 ms, which can cause a spike in the latency. We should
-        # consider moving this to a separate thread.
+
+        # Apply multi-modal mapper (if necessary)
         if req.mm_data:
+            assert req.mm_inputs is None or req.mm_inputs == []
```

Comment:
Shouldn't `req.mm_inputs` here always be a `List[MultiModalKwargs]`? Why do we need the `req.mm_inputs is None` here?

Reply:
Removed